### DISCUSSION TITLE
MIP - Versioning & CAIP-27

### DISCUSSION THREAD

**Discussion by vandan** [MetaMask/metamask-improvement-proposals#12] [MIPs]
In this [PR](https://github.com/MetaMask/metamask-improvement-proposals/pull/11), should we specify whether dapps can use multiple versions of the API in tandem as long as they are not deprecated?

**Comment by shanejonas** [comment]
I think it makes sense for dapp developers to upgrade when their own timeline allows (within reason, within deprecation period).

A product team might have a large codebase and could take a while to upgrade breaking changes.
